File: Challenge/5993. Keep Multiplying Found Values by Two.py
# https://leetcode.com/contest/weekly-contest-278/problems/keep-multiplying-found-values-by-two/
class Solution:
    def findFinalValue(self, nums: List[int], original: int) -> int:
        
        c = collections.Counter(nums)
        
        while True:
            if c.get(original, False):
                original *= 2
            else:
                return original

File: Challenge/5993. Keep Multiplying Found Values by Two.cpp
// https://leetcode.com/contest/weekly-contest-278/problems/keep-multiplying-found-values-by-two/
class Solution {
public:
    int findFinalValue(vector<int>& nums, int original) {
        
        set<int> s;
        
        for (const auto &i : nums){
            s.insert(i);
        }
        while (true){
            auto it = s.find(original);
            if (it != s.end()){
                original *= 2;
            } else {
                return original;
            }
        }
    }
};
